add bench armrests quest

newplaylist47 opened the pull request:
Fixes https://github.com/Helium314/SCEE/issues/834

newplaylist47 committed:
Create AddBenchArmrest.kt

newplaylist47 committed:
Create BenchArmrestForm.kt

newplaylist47 committed:
Rename BenchArmrestForm.kt to AddBenchArmrestForm.kt

newplaylist47 committed:
Create BenchArmrestAnswer.kt

newplaylist47 committed:
Add bench_armrest into QuestsModule.kt

newplaylist47 committed:
add AddBenchArmrest() to QuestsModule.kt

newplaylist47 committed:
add bench armrest string to strings_ee.xml

mnalis commented on app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/bench_armrest/AddBenchArmrest.kt:
```diff
@@ -0,0 +1,50 @@
+    """
+    override val changesetComment = "Survey whether benches have armrests"
+    override val wikiLink = "Tag:amenity=bench"
+    override val icon = R.drawable.ic_quest_bench_poi
```
Do we perhaps need (slightly) different icon? 
It might be somewhat disorienting/confusing if we solve backrest quest, and then the same icon pops up (with very similar text)...

Anyone good with vector icons?

mnalis commented on app/src/androidMain/res/values/strings_ee.xml:
```diff
@@ -235,6 +235,7 @@
 
 
     <!-- new quests -->
+    <string name="quest_bench_armrest_title">Does this bench have armrests?</string>
```
```suggestion
    <string name="quest_bench_armrest_title">Does this bench have any armrest?</string>
```

or something similar... because, according to to wiki, **single** armrest also makes it eligible for `yes` answer.

Also,  I'd move it close to existing `<!-- bench material -->` string (and add a `<!-- bench armrests -->` heading)

mnalis reviewed (commented):
Congrats on your first PR @newplaylist47 ! 

I'm glad to hear debug APK worked and quest behaves as expected (if I understood correctly)

Could you change/move that string? 

Other that that (and icon, but do not worry about it if it is not your forte) it looks good to me!

(P.S. I've also changed the text to "Fixes" as that is magic word that GitHub uses to automatically close the issue when PR is merged).

mnalis committed:
Apply suggestion from @mnalis

change wording; as (according to the wiki) single armrest also makes it eligible for `yes` answer

mnalis commented on app/src/androidMain/res/values/strings_ee.xml:
```diff
@@ -235,6 +235,7 @@
 
 
     <!-- new quests -->
+    <string name="quest_bench_armrest_title">Does this bench have any armrest?</string>
```
Could you just move this string closer to existing `<!-- bench material -->` string (and add a `<!-- bench armrests -->` heading) ?

newplaylist47 committed:
moved bench armrest quest next to bench materials quests

as per suggestion https://github.com/Helium314/SCEE/pull/847#pullrequestreview-3767812765

mnalis commented on app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/QuestsModule.kt:
```diff
@@ -611,6 +612,7 @@ fun getQuestTypeList(
     154 to AddWayLit(),
 
     // quests added in SCEE
+    EE_QUEST_OFFSET + 58 to AddBenchArmrest(),
```
```suggestion
    EE_QUEST_OFFSET + 59 to AddBenchArmrest(),
```

58 was taken by recent `AddOnewayBicycle` (says `fgrep EE_QUEST_OFFSET **/QuestsModule.kt  | sort -g -k+3`)

mnalis committed:
Update app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/QuestsModule.kt

bump quest id

newplaylist47 committed:
add bench_armrest svg

newplaylist47 committed:
add bench armrest drawable

newplaylist47 committed:
update icon value

Helium314 commented:
Looks good, thanks!
